### PR TITLE
switch to rebranded blake2

### PR DESCRIPTION
**Pull request by matklad**
This pulls
https://github.com/near/near-blake2/commit/cebb7df79608a7058017940830d37c37d55d21fe
which just renames blake2. It's parent, 736ff60, is what we are using
currently. There are some extra commits on top of that in master, but I
decided not to pull them in just yet, to minimize change.

The primary motivation here is making it easy to upgrade other things:

   blake2 = { git = "https://github.com/near/near-blake2.git", version = "0.9.1", default-features = false }

makes that tricky, as there's no longer `blake2` package at that URL!

### CODE REVIEW

**Review by birchmd — APPROVED** [review]
LGTM!

**Review by joshuajbouw — APPROVED** [review]
Thanks, been meaning to get around to this eventually. Will fix out libs that use the aurora-engine without touching Cargo.lock manually hah!

**Comment by matklad** [comment]
Is there anything I should do to make this mergable?

**Comment by birchmd** [comment]
I guess if you don't see the option to merge it could be due to the repo permissions. I'll merge all your approved PRs now.

**Comment by matklad** [comment]
Yeah, I don't see the option (and I probably shouldn't: I'd rather be a contributor than a maintainer :) ), so I assumed it was waiting on some extra review. Should've pinged earlier, sorry about that!

**Pull request merged**
